lib/cfa/augeas_parser/writer.rb

```diff
@@ -306,6 +306,11 @@ def path_after(preceding)
         # e.g. #comment to #comment[1]. Can happen only if it switch from
         # single entry to collection
         preceding_index ||= paths.index(preceding.path + "[1]")
+        # it can also happen that prefix is change from single entry to collection
```

Comment:
For the particular issue would be ok, although would be nice to have the recursive check (general solution ASAP).